src/main/java/uk/gov/hmcts/darts/authentication/dao/impl/AzureDaoImpl.java

```diff
@@ -0,0 +1,72 @@
+package uk.gov.hmcts.darts.authentication.dao.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import feign.Response;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.util.StreamUtils;
+import uk.gov.hmcts.darts.authentication.config.AuthenticationConfiguration;
+import uk.gov.hmcts.darts.authentication.dao.AzureDao;
+import uk.gov.hmcts.darts.authentication.exception.AzureDaoException;
+import uk.gov.hmcts.darts.authentication.model.OAuthProviderRawResponse;
+import uk.gov.hmcts.darts.authentication.service.AzureActiveDirectoryB2CClient;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor
+public class AzureDaoImpl implements AzureDao {
+
+    private final AuthenticationConfiguration authConfig;
+    private final AzureActiveDirectoryB2CClient azureActiveDirectoryB2CClient;
+
+    @Override
+    public OAuthProviderRawResponse fetchAccessToken(String code) throws AzureDaoException {
+        log.debug("Fetching access token(s) for authorization code: {}", code);
+
+        if (StringUtils.isBlank(code)) {
+            throw new AzureDaoException("Null code not permitted");
+        }
+
+        MultiValueMap<String, String> requestBody = new LinkedMultiValueMap<>();
+        requestBody.add("grant_type", authConfig.getExternalADauthorizationGrantType());
+        requestBody.add("redirect_uri", authConfig.getExternalADredirectUri());
+        requestBody.add("code", code);
+        requestBody.add("client_id", authConfig.getExternalADclientId());
+        requestBody.add("client_secret", authConfig.getExternalADclientSecret());
+        requestBody.add("resource", authConfig.getExternalADclientId());
```

Comment:
Can you check this field is needed in the request body? I couldn't see it myself
https://learn.microsoft.com/en-gb/azure/active-directory-b2c/access-tokens